api-requests.js

```diff
@@ -0,0 +1,75 @@
+const request = require('request-promise-native');
+const moment = require('moment');
+
+// API Credentials
+const appId = '4586bfa0';
+const appKey = '4236304488967b5683897c8980725377';            
+
+function getBusArrivalJSON(postcode) {
+    
+    return request(`https://api.postcodes.io/postcodes/${postcode}`)
+
+    .then(body => {
+        const postcodeInfoAsJSON = JSON.parse(body);
+        
+        // Extract postcode lat lon
+        const latitude = postcodeInfoAsJSON['result']['latitude'];
+        const longitude = postcodeInfoAsJSON['result']['longitude'];
+        
+        const radius = 1000; 
+        const busStopURL = `https://api.tfl.gov.uk/StopPoint?stopTypes=NaptanOnstreetBusCoachStopPair&radius=${radius}&lat=${latitude}&lon=${longitude}`;   
+        return request(busStopURL);
+    })
+    
+    .catch(err => {
+        // Both 404 status rejections and invalid postcodes end up in this catch block
+        if (err.name === 'StatusCodeError') {
+            return Promise.reject('Invalid post code');
+        } else {
+            return Promise.reject('Couldn\'t reach api.postcodes.io');
+        }
+    })
+
+    .then(body => {
+        const stopPointsAsJSON = JSON.parse(body);
+        let promises = [];
+
+        // Get data from first two closest bus stops
+        stopPointsAsJSON['stopPoints'].slice(0,2).forEach(function(stop) {
+            stop['lineGroup'].forEach(function(directedStop) {
+                const stopId = directedStop['naptanIdReference'];
+                const requestURL = `https://api.tfl.gov.uk/StopPoint/${stopId}/Arrivals?app_id=${appId}&app_key=${appKey}`;
+
+                // Get bus times from this stop
+                promises.push(request(requestURL));
+            });
+        });
+        return Promise.all(promises); // Turn an array of promises into a promise of an array
+    })
+
+    .then(bodies => {

```

Comment:
```javascript
bodies => bodies.map(buildBusTimesJSON)
```

The extra `Promise.resolve` doesn't add anything (it'll just be unwrapped since we're in a `then` block.

Your `buildBusTimesJSON` is already a function that takes the body and returns your bus times; no need for the arrow function (instead of passing `x => f(x)` just pass `f`). This change isn't always possible (it'll break if you try it with `console.log`, for example), but in this case you wrote the function and you know it takes exactly one argument and doesn't do anything fancy with it, so this works fine.